statsd/statsd.go

```diff
@@ -71,7 +83,14 @@ var (
 // A Client is a handle for sending udp messages to dogstatsd.  It is safe to
 // use one Client from multiple goroutines simultaneously.
 type Client struct {
+	// Address to send metrics to, needed to allow reconnection on error
+	addr net.Addr
+	// Established connection object, or nil if Connect needed
 	conn net.Conn
+	// function pointer to use to write data to the connection
+	write func([]byte) error
```

Comment:
On a second thought, it would be more idiomatic to have this be an interface to `io.Writer`.

And have two different structs, one for UDP and one for UDS, both implementing `Writer`. We can assign the right one during the creation of the `Client`. Should be straight-forward.